edb/schema/indexes.py

```diff
@@ -356,6 +361,12 @@ class Index(
         allow_ddl_set=True,
     )
 
+    # If this is a generated AI index, track the created type
+    generated_ai_model_type = so.SchemaField(
```

Comment:
Is this just to keep the reference?  I wonder if you can formulate the generated ref injection via `find_extra_refs` callback in `compile_expr_field` instead?